File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int i =0;
        int j= h.size()-1;
        int maxi = INT_MIN;
        while(i<j){
            int w = j-i;
            int hi = min(h[j],h[i]);
            maxi  = max(maxi,w*hi);
            if(h[j]<h[i]){
                j--;
            }
            else{
                i++;
            }
        }
        return maxi;
    }
};

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var ans =0;
    let i=0,j=height.length-1;
    while(i<j){
        let temp = Math.min(height[i],height[j])*(j-i);
        ans = Math.max(ans,temp);
        if(height[i]>height[j]){
            j--;
        }
        else{
            i++;
        }
    }
    return ans;

};